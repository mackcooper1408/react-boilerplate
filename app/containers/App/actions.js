/*
 * App Actions
 */

import {
  LOAD_STRINGS,
  LOAD_STRINGS_SUCCESS,
  LOAD_STRINGS_ERROR,
  ADD_STRING,
  ADD_STRING_SUCCESS,
  ADD_STRING_ERROR,
} from './constants';

/**
 * Load the strings, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_STRINGS
 */
export function loadStrings() {
  return {
    type: LOAD_STRINGS,
  };
}

/**
 * Dispatched when the strings are loaded by the request saga
 *
 * @param  {array} strings The strings data
 *
 * @return {object}      An action object with a type of LOAD_STRINGS_SUCCESS passing the strings
 */
export function stringsLoaded(strings) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    strings,
  };
}

/**
 * Dispatched when loading the strings fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of LOAD_STRINGS_ERROR passing the error
 */
export function stringsLoadingError(error) {
  return {
    type: LOAD_STRINGS_ERROR,
    error,
  };
}

/**
 * add a string, this action starts the request saga
 *
 * @return {object} An action object with a type of ADD_STRING
 */
export function addString() {
  return {
    type: ADD_STRING,
  };
}

/**
 * Dispatched when the string is added by the request saga
 *
 * @param  {array} strings The strings data
 *
 * @return {object}      An action object with a type of ADD_STRING_SUCCESS passing the strings
 */
export function stringAdded(stringData) {
  return {
    type: ADD_STRING_SUCCESS,
    stringData,
  };
}

/**
 * Dispatched when adding a string fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of ADD_STRING_ERROR passing the error
 */
export function stringAddedError(error) {
  return {
    type: ADD_STRING_ERROR,
    error,
  };
}
