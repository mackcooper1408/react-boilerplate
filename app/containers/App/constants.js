/*
 * App Constants
 */

export const LOAD_STRINGS = 'dmi-tech-screen/App/LOAD_STRINGS';
export const LOAD_STRINGS_SUCCESS = 'dmi-tech-screen/App/LOAD_STRINGS_SUCCESS';
export const LOAD_STRINGS_ERROR = 'dmi-tech-screen/App/LOAD_STRINGS_ERROR';
export const ADD_STRING = 'dmi-tech-screen/App/ADD_STRING';
export const ADD_STRING_SUCCESS = 'dmi-tech-screen/App/ADD_STRING_SUCCESS';
export const ADD_STRING_ERROR = 'dmi-tech-screen/App/ADD_STRING_ERROR';
