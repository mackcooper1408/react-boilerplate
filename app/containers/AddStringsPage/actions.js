/*
 * Add Strings Page actions
 */

import { CHANGE_STRING_DATA } from './constants';

/**
 * Changes the input field of the form
 *
 * @param  {string} stringData The new text of the input field
 *
 * @return {object} An action object with a type of CHANGE_STRING_DATA
 */
export function changeStringData(string) {
  return {
    type: CHANGE_STRING_DATA,
    string,
  };
}
