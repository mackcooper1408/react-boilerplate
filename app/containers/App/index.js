/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */

import React from 'react';
import { Switch, Route } from 'react-router-dom';

import HomePage from 'containers/HomePage/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';
import StringsPage from 'containers/StringsPage/Loadable';
import AddStringsPage from 'containers/AddStringsPage/Loadable';
import { v4 as uuid } from 'uuid';

import GlobalStyle from '../../global-styles';
import Navbar from '../../components/Navbar';

const navItems = [
  { id: uuid(), text: 'Strings', link: '/strings' },
  { id: uuid(), text: 'Add', link: '/add' },
];

export default function App() {
  return (
    <div>
      <Navbar navItems={navItems} />
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route exact path="/strings" component={StringsPage} />
        <Route exact path="/add" component={AddStringsPage} />
        <Route component={NotFoundPage} />
      </Switch>
      <GlobalStyle />
    </div>
  );
}
