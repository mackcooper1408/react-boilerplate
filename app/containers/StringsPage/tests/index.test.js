import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { StringsPage, mapDispatchToProps } from '../index';

import { loadStrings } from '../../App/actions';
import configureStore from '../../../configureStore';

describe('<StringsPage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <StringsPage loading={false} error={false} strings={[]} />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should fetch the strings on mount', () => {
    const mockOnLoad = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <StringsPage onLoad={mockOnLoad} />
        </IntlProvider>
      </Provider>,
    );
    expect(mockOnLoad).toHaveBeenCalled();
  });

  it('should not call onLoad if store contains strings', () => {
    const strings = [{ id: '1', item: 'test1' }, { id: '2', item: 'test2' }];
    const mockOnLoad = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <StringsPage onLoad={mockOnLoad} strings={strings} />
        </IntlProvider>
      </Provider>,
    );
    expect(mockOnLoad).not.toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    describe('onLoad', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onLoad).toBeDefined();
      });

      it('should dispatch loadStrings when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onLoad();
        expect(dispatch).toHaveBeenCalledWith(loadStrings());
      });
    });
  });
});
