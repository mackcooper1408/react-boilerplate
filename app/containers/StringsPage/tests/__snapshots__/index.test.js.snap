// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StringsPage /> should render and match the snapshot 1`] = `
<div>
  <h1
    class="H1-tncdg6-0 gNmgOQ"
  >
    <span>
      Here are all the strings!
    </span>
  </h1>
  <div
    class="Wrapper-sc-1umgotm-0 kPmOOE"
  >
    <ul
      class="Ul-mwnq6h-0 cbNGBL"
    />
  </div>
</div>
`;
